### PR TITLE
332 - GlobalVar - GroupGenerator

### LINKED ISSUE #332: [200 SigmaUSD] Implement `groupGenerator` global var
- Add enum variant `GroupGenerator` to `ergotree-ir::mir::global_vars::GlobalVars`;
- Write serialization roundtrip tests for all `GlobalVars` variants (PBT will be suffice, see `serialization::expr::tests::ser_roundtrip`);
- Add `GlobalGenerator` enum variant handling in `ergotree-interpreter::eval::global_vars`;

Reference Scala version:
IR node and eval - https://github.com/ScorexFoundation/sigmastate-interpreter/blob/b60edae20f2af3352e4064d139d45a15fc507f7d/sigmastate/src/main/scala/sigmastate/Values.scala#L691-L701

### PR DESCRIPTION
**Pull request by ApexTheory**
GroupGenerator added to global vars.  I believe I have this correct, to use the dlog_group::generator?

EDIT(@greenhat ): Close #332

### CODE REVIEW

File: ergotree-interpreter/src/eval/global_vars.rs [hunk 1]
```diff
@@ -68,4 +71,11 @@ mod tests {
             ctx.inputs
         );
     }
+
+    #[test]
+    fn eval_group_generator() {
+        let ctx = Rc::new(force_any_val::<Context>());
+        let res = eval_out::<EcPoint>(&GlobalVars::GroupGenerator.into(), ctx.clone());
+        assert!(matches!(res.into(), Value::GroupElement(_)));
```
Comment (greenhat) [inline ergotree-interpreter/src/eval/global_vars.rs:79]: For regression test checking that evaluation returns an exact value (`dlog_group::generator()`) would be a better fit.

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking good! Thanks! Just one comment about the test.

File: ergotree-interpreter/src/eval/global_vars.rs (hunk 1 above)
Comment (ApexTheory) [reply ergotree-interpreter/src/eval/global_vars.rs]: Fixed.

**Review by greenhat — APPROVED** [review]
Thanks!

**Pull request merged**